packages/datadog-instrumentations/src/helpers/instrument.js

```diff
@@ -28,12 +30,22 @@ exports.addHook = function addHook ({ name, versions, file }, hook) {
       return moduleExports
     }
 
-    try {
-      return hook(moduleExports)
-    } catch (e) {
-      log.error(e)
-      return moduleExports
+    if (moduleName === fullFilename) {
```

Comment:
wouldn't `moduleName` already be equal to `fullFilename` here? 

Would this work?
```javascript
const version = getVersion(moduleBaseDir)
if (!matchVersion(version, versions)) {
  return moduleExports
}
try {
  loadChannel.publish({ name, version, file })

  return hook(moduleExports)
} catch (e) {
  log.error(e)
  return moduleExports
}
```

Reply:
actually the version would also match I think, so we don't even need the `matchVersion` check I think 🤔

Reply:
I meant to move the code but instead I just copied, fixing now.